nova/compute/api.py

```diff
@@ -695,6 +695,33 @@ def _merge_bdms_lists(self, overridable_mappings, overrider_mappings):
                 [bdm for bdm in overridable_mappings
                  if bdm['device_name'] not in device_names])
 
+    def _flavor_enforces_boot_from_volume(self, flavor):
+        """Determines whether boot-from-volume is enforced by flavor"""
+        return (flavor.get('extra_specs', {})
+                .get('boot_from_volume', 'false') == 'true')
+
+    def _enforce_boot_from_volume(self, image_meta, flavor,
+                                  block_device_mapping):
+        """Adds boot-from-volume block device mapping if not already set."""
+        root_bdm = block_device.get_root_bdm(block_device_mapping)
+        if root_bdm:
+            if root_bdm.get('destination_type') == 'volume':
+                # User already set a root volume to boot from.
+                return
+            elif (root_bdm.get('destination_type') == 'local' and
+                  root_bdm.get('source_type') == 'image'):
+                # In this case overwrite the image mapping to a volume.
```

Comment:
This comment feels redundant and/or possibly misleading. I guess it's supposed to describe what's happening in _all_ the rest of the function, but it looks like it's only describing the next line.

The code does describe itself well enough without this comment.

The other comment above is not quite as obvious, but the early `return` is a pretty strong indicator of "we're done early here". Personally, I'd also remove it.